tests/components/xiaomi_miio/test_fan.py

```diff
@@ -0,0 +1,127 @@
+"""The tests for the xiaomi_miio fan component."""
+
+from collections.abc import Generator
+from unittest.mock import MagicMock, Mock, patch
+
+from miio.integrations.fan.dmaker.fan import FanStatusP5
+from miio.integrations.fan.dmaker.fan_miot import FanStatusMiot
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.xiaomi_miio import MODEL_TO_CLASS_MAP
+from homeassistant.components.xiaomi_miio.const import CONF_FLOW_TYPE, DOMAIN
+from homeassistant.const import (
+    CONF_DEVICE,
+    CONF_HOST,
+    CONF_MAC,
+    CONF_MODEL,
+    CONF_TOKEN,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from . import TEST_MAC
+
+from tests.common import MockConfigEntry, snapshot_platform
+
+
+@pytest.fixture(
+    name="model_code",
+    params=[
+        "dmaker.fan.p5",
+        "dmaker.fan.p18",
+    ],
+)
+def get_model_code(request: pytest.FixtureRequest) -> str:
+    """Parametrize model code."""
+    return request.param
+
+
+@pytest.fixture(autouse=True)
+def setup_device(model_code: str) -> Generator[MagicMock]:
+    """Initialize test xiaomi_miio for fan entity."""
+
+    mock_fan = MagicMock()
+
+    if model_code == "dmaker.fan.p5":
```

Comment:
we usually avoid having conditions in tests.
alternatively, we could have 2 methods (`get_fan_p5_mock` + `get_fan_p18_mock` or something like that), which return the data dict, and then have a tuple in the fixture params `(model_code, path_class, mock_status)`.

Reply:
> we usually avoid having conditions in tests. alternatively, we could have 2 methods (`get_fan_p5_mock` + `get_fan_p18_mock` or something like that), which return the data dict, and then have a tuple in the fixture params `(model_code, path_class, mock_status)`.

I fully understand avoiding conditions in tests. However, this is a fixture and I thought it made sense here...

I'll see if I can refactor